dune_client/api/extensions.py

```diff
@@ -76,35 +80,37 @@ def run_query(
 
         # pylint: disable=duplicate-code
         job_id = self._refresh(query, ping_frequency, performance)
+        params = GetExecutionResultsParams(
+            limit, columns, sample_count, filters, sort_by, None
+        )
         return self._fetch_entire_result(
             self.get_execution_results(
                 job_id,
-                columns=columns,
-                sample_count=sample_count,
-                filters=filters,
-                sort_by=sort_by,
-                limit=limit,
                 allow_partial_results=allow_partial_results,
+                params=params,
             ),
         )
 
     def run_query_csv(
         self,
         query: QueryBase,
         ping_frequency: int = POLL_FREQUENCY_SECONDS,
-        performance: Optional[str] = None,
-        batch_size: Optional[int] = None,
-        columns: Optional[List[str]] = None,
-        sample_count: Optional[int] = None,
-        filters: Optional[str] = None,
-        sort_by: Optional[List[str]] = None,
+        params: Optional[Dict[str, Any]] = None,
     ) -> ExecutionResultCSV:
         """
         Executes a Dune query, waits till execution completes,
         fetches and the results in CSV format
         (use it load the data directly in pandas.from_csv() or similar frameworks)
         """
         # Ensure we don't specify parameters that are incompatible:
+        if params is None:
+            params = {}
+        performance = params.get("performance", None)
+        batch_size = params.get("batch_size", None)
+        columns = params.get("columns", None)
+        sample_count = params.get("sample_count", None)
+        filters = params.get("filters", None)
+        sort_by = params.get("sort_by", None)
```

Comment:
There is way too much redundant code here. All of this appears too many times over.